CAMEL-8941 improved restlet binary data handling

koscejev opened the pull request:
Additional support for camel-restlet to properly handle binary data:
* consumer can return binary data as InputStream (test only)
* consumer can return binary data as byte\[\] (test + fix)
* producer can receive "application/octet-stream" content as InputStream (test only)
* producer can receive other content as InputStream (test + fix)

davsclaus commented:
Thanks for this PR but there is a bunch of unit test errors. Do you not have those?

koscejev commented:
Apologies, I manually ran only tests that I changed myself and executed maven build to verify tests that I didn't edit. Turns out surefire tests are disabled by default on Windows.

The tests that are failing are explicitly casting the result body to String. I've modified these tests to request body of String via Camel, so that Camel can use converters if necessary. Do you think this is appropriate or should the returned body be String without any conversions for all existing cases?

Also, I managed to make tests work on Windows, so I propose removing the windows profile that disables tests.

If you find that only some changes are appropriate, feel free to only take the commits you're OK with.

davsclaus commented:
Thanks for the updates. The code has been merged. Do you mind closing this?
